dali/operators/nvcvop/nvcvop.h

```diff
@@ -0,0 +1,183 @@
+    arg.Acquire(spec_, ws, num_samples, arg_shape);
+    T *staging_buffer =
+        scratchpad.AllocatePinned<T>(num_samples * arg_sample_vol, dtype.alignment());
+    for (int s = 0; s < num_samples; ++s) {
+      std::copy(arg[s].data, arg[s].data + arg_sample_vol,
+                staging_buffer + s * arg_sample_vol);
+    }
+    T *device_buffer = scratchpad.AllocateGPU<T>(num_samples * arg_sample_vol, dtype.alignment());
+    MemCopy(device_buffer, staging_buffer, num_samples * arg_sample_vol * sizeof(T), ws.stream());
+    int ndim = arg_shape.sample_dim();
+    auto inner_dim = arg_shape[ndim - 1];
+    DALI_ENFORCE(inner_dim % dtype.numChannels() == 0, "Invalid argument shape.");
+    std::vector<int64_t> shape_data(ndim);
+    shape_data[0] = num_samples;
+    for (int d = 0; d < ndim; ++d) {
+      shape_data[d + 1] = arg_shape[d];
+    }
+    if (inner_dim != dtype.numChannels()) {
+      shape_data.push_back(inner_dim / dtype.numChannels());
+    }
+    nvcv::TensorDataStridedCuda::Buffer inBuf;
+    inBuf.basePtr = reinterpret_cast<NVCVByte*>(device_buffer);
+    inBuf.strides[shape_data.size() - 1] = dtype.strideBytes();
+    for (int d = shape_data.size() - 2; d >= 0; --d) {
+      inBuf.strides[d] = shape_data[d + 1] * inBuf.strides[d + 1];
+    }
+    nvcv::TensorShape shape(shape_data.data(), shape_data.size(), nvcv::TensorLayout(""));
+    nvcv::TensorDataStridedCuda inData(shape, dtype, inBuf);
+    return nvcv::TensorWrapData(inData);
+  }
+
+  /**
+   * @brief Get input image batch.
+   * This method allocates the ImageBatchVarShape and wraps the input data as nvcv images.
+   */
+  const nvcv::ImageBatchVarShape &GetInputBatch(Workspace &ws, size_t input_idx) {
+    if (inputs_.size() < input_idx + 1) {
+      inputs_.resize(input_idx + 1, nvcv::ImageBatchVarShape{1});
+    }
+    const auto &tl_input = ws.Input<GPUBackend>(input_idx);
+    auto &input = inputs_[input_idx];
+    if (input.capacity() < tl_input.num_samples()) {
+      input = nvcv::ImageBatchVarShape(std::max(input.capacity() * 2, tl_input.num_samples()));
+    }
+    input.clear();
+    PushImagesToBatch(tl_input, input);
+    return input;
+  }
+
+  /**
+   * @brief Get output image batch.
+   * This method allocates the ImageBatchVarShape and wraps the input data as nvcv images.
+   */
+  nvcv::ImageBatchVarShape &GetOutputBatch(Workspace &ws, size_t output_idx) {
+    if (outputs_.size() < output_idx + 1) {
+      outputs_.resize(output_idx + 1, nvcv::ImageBatchVarShape{1});
+    }
+    const auto &tl_output = ws.Output<GPUBackend>(output_idx);
+    auto &output = outputs_[output_idx];
+    if (output.capacity() < tl_output.num_samples()) {
+      output = nvcv::ImageBatchVarShape(std::max(output.capacity() * 2, tl_output.num_samples()));
+    }
+    output.clear();
+    PushImagesToBatch(tl_output, output);
+    return output;
+  }
+
+  std::vector<nvcv::ImageBatchVarShape> inputs_;
+  std::vector<nvcv::ImageBatchVarShape> outputs_;
```

Comment:
This is dangerous. The image batch objects (as of now) are ephemeral - they only live as long as the workspace is valid. As such, we should not extend their lifetime beyond the Run function. I propose that we implement a small `RunImpl` function here which clears those vectors at the end.

Reply:
I implemented the Run() method that cleans the batches after the RunImpl